src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Globalization.cs

```diff
@@ -107,28 +101,22 @@ public static int CompareTo(this ReadOnlySpan<char> span, ReadOnlySpan<char> oth
             switch (comparisonType)
             {
                 case StringComparison.CurrentCulture:
-                    return CultureInfo.CurrentCulture.CompareInfo.CompareOptionNone(span, other);
-
                 case StringComparison.CurrentCultureIgnoreCase:
-                    return CultureInfo.CurrentCulture.CompareInfo.CompareOptionIgnoreCase(span, other);
+                    return CultureInfo.CurrentCulture.CompareInfo.Compare(span, other, string.GetCaseCompareOfComparisonCulture(comparisonType));
 
                 case StringComparison.InvariantCulture:
-                    return CompareInfo.Invariant.CompareOptionNone(span, other);
-
                 case StringComparison.InvariantCultureIgnoreCase:
-                    return CompareInfo.Invariant.CompareOptionIgnoreCase(span, other);
+                    return CompareInfo.Invariant.Compare(span, other, string.GetCaseCompareOfComparisonCulture(comparisonType));
 
                 case StringComparison.Ordinal:
                     if (span.Length == 0 || other.Length == 0)
                         return span.Length - other.Length;
                     return string.CompareOrdinal(span, other);
 
-                case StringComparison.OrdinalIgnoreCase:
+                default:
```

Comment:
why you have changed this to default? I see you assert but not necessary can catch a problem in release builds.